codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/S3100Continue.java

```diff
@@ -0,0 +1,77 @@
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import java.util.List;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoDelegator;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.MiddlewareRegistrar;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.utils.ListUtils;
+
+/**
+ * Add middleware, which adds {Expect: 100-continue} header for s3 client HTTP PUT request larger than 2MB
+ * or with unknown size streaming bodies, during operation builder step
+ */
+public class S3100Continue implements GoIntegration {
```

Comment:
**fix**: Not sure where to make this comment but we ran into an issue with this header being set and signing. See this [PR](https://github.com/awslabs/smithy-kotlin/pull/812) for more details. Can you verify that s3 put object requests with transfer acceleration enabled don't run into issues when this header is present? We may need to special case it in signing as we did in Kotlin.

Reply:
I'm confused about the test design of this validation. Since  `UseAccelerate` and `SupportsAccelerate` is configured in [UpdateEndpoint](https://github.com/aws/aws-sdk-go-v2/blob/main/service/s3/internal/customizations/update_endpoint.go#:~:text=func-,UpdateEndpoint,-(stack%20*) at Serialize step in stack, is it possible to wire up inner middleware like `processARNResource`, `s3ObjectLambdaEndpoint` etc with current `s3100Continue` middleware in isolation in unit test to check issues?

Reply:
I'm confused about the test design of this validation. Since UseAccelerate and SupportsAccelerate is configured in [UpdateEndpoints](https://github.com/aws/aws-sdk-go-v2/blob/main/service/s3/internal/customizations/update_endpoint.go#:~:text=func-,UpdateEndpoint,-(stack%20*) at Serialize step in stack, is it possible to wire up inner middleware like processARNResource, s3ObjectLambdaEndpoint etc with current s3100Continue middleware in isolation in unit test to check issues?

Reply:
I'm confused about the test design of this validation. Since UseAccelerate and SupportsAccelerate is configured in [UpdateEndpoints](https://github.com/aws/aws-sdk-go-v2/blob/main/service/s3/internal/customizations/update_endpoint.go#:~:text=func-,UpdateEndpoint,-(stack%20* ) at Serialize step in stack, is it possible to wire up inner middleware like `processARNResource`, `s3ObjectLambdaEndpoint` etc with current s3100Continue middleware in isolation in unit test to check issues?

Reply:
No need to check it in a unit test. Just test the actual real S3 service with your customization enabled and `UseAccelerate` enabled (might need to [configure](https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration-examples.html) it on the bucket).